posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -254,10 +255,147 @@ More information about Expression Language 5.0 can be found in:
 
 * link:{url-prefix}/docs/latest/reference/feature/expressionLanguage.html[Open Liberty Expression Language feature documentation]
 
+[#concurrency]
+=== Concurrency 3.0 for Jakarta EE 10
+
+The Jakarta Concurrency specification enhances the set of parallel and reactive programming models available to applications, enabling tasks and completion stage actions to run with the thread context of the submitter.
+
+Concurrency 3.0 includes the following enhancements:
+
+* Resource definition annotations that put applications in control of defining the concurrency constraints and thread context types that the application requires.
+* Context-aware completion stages that are managed by the container.
+* Asynchronous methods that are backed by context-aware completion stages.
+* Improvements to Triggers for customizing the scheduling of business logic according to dates and times within one's own time zone.
+* Standardized integration for third-party providers of thread context.
+
+Enable the concurrent-3.0 feature in server.xml, along with other Jakarta EE 10 beta features that you wish to use in combination with it, for example:
+
+[source, xml]
+----
+<featureManager>
+  <feature>concurrent-3.0</feature>
+  <!-- other features that are used in the examples: -->
+  <feature>cdi-4.0</feature>
+  <feature>jdbc-4.2</feature>
+  <feature>jndi-1.0</feature>
+  <feature>servlet-6.0</feature>
+</featureManager>
+----
+
+You can still configure `<managedExecutorService>` and other Concurrency resource types in your `server.xml` file as you did for previous versions of the feature.
+
+However, Concurrency 3.0 gives you the ability to define those resources within application components. For example:
+
+[source, java]
+----
+import static jakarta.enterprise.concurrent.ContextServiceDefinition.ALL_REMAINING;
+import static jakarta.enterprise.concurrent.ContextServiceDefinition.APPLICATION;
+import static jakarta.enterprise.concurrent.ContextServiceDefinition.SECURITY;
+import static jakarta.enterprise.concurrent.ContextServiceDefinition.TRANSACTION;
+
+import jakarta.annotation.Resource;
+import jakarta.enterprise.concurrent.ContextServiceDefinition;
+import jakarta.enterprise.concurrent.ManagedExecutorDefinition;
+import jakarta.enterprise.concurrent.ManagedExecutorService;
+import jakarta.inject.Inject;
+import jakarta.servlet.http.*;
+
+import java.sql.*;
+import java.util.concurrent.CompletionException;
+
+@ContextServiceDefinition(name = "java:app/concurrent/AppContextOnly",
+        propagated = APPLICATION,
+        cleared = { TRANSACTION, SECURITY },
+        unchanged = ALL_REMAINING)
+@ManagedExecutorDefinition(name = "java:module/concurrent/MyExecutor",
+        context = "java:app/concurrent/AppContextOnly",
+        maxAsync = 5)
+public class MyServlet extends HttpServlet {
```

Comment:
Servlet is not available.  Should it be changed to a RESTful Web Services resource instead?

Reply:
Since servlet 6.0 isn't available is there even value with updating the concurrency section in the beta blog?  @njr-11 / @KyleAure ?

Reply:
Given that the Concurrency examples are all written for Servlet, I recommend commenting them out in the beta for now, starting with the text "For example:" and everything after that.  In a future beta when servlet-6.0 is added, then uncomment them.

Reply:
```suggestion
You can still configure `<managedExecutorService>` and other Concurrency resource types in your `server.xml` file as you did for previous versions of the feature.  However, In Concurrency 3.0 you can define those resources within application components.
```